src/main/java/org/jenkinsci/plugins/workflow/libs/LibraryAdder.java

```diff
@@ -85,7 +85,19 @@
             librariesUnparsed.put(parsed[0], library);
         }
         List<Addition> additions = new ArrayList<>();
-        LibrariesAction action = build.getAction(LibrariesAction.class);
+        LibrariesAction action = null;
+        
+        for( LibrariesAction laction : build.getActions( LibrariesAction.class ) ) {
+            if( laction.getScope() != null && laction.getScope().equals( scope ) ) {
+                // found the action that's absolutely right for this scope
+                action = laction;
+                break;
+            } else if( laction.getScope() == null ) {
+                // generic LibrariesAction may be overridden later if there's a perfect match.
+                action = laction;
```

Comment:
It is possible that a build was suspended before these code changes landed, in which case the "right" LibrariesAction will have a `null` scope.